### PR TITLE
(#874) Uswds Table Component

### LINKED ISSUE #874: Table USWDS Pass Through to NCIDS
### USWDS Component Name
[Table](https://designsystem.digital.gov/components/table/)

### USWDS Component Packages
* Package usage: `@forward "usa-table";`
* Dependencies: `uswds-fonts</code class="text-no-wrap"> </li> </ul>`

### Subtasks
- [x] Make a folder with the component name under uswds-native
- [x] Copy USWDS twig and any JSON
- [x] Make an index.scss for the stories
- [x] Copy and hack up USWDS stories.js file to use our story format
- [x] Create scenarios.js by adding a scenario for each story in stories.js
- [x] Add link to the USWDS documentation on the NCIDS documentation site

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #874

### CODE REVIEW

**Comment by karlikpj** [comment]
[Storybook](https://designsystem-dev.cancer.gov/pr-906/storybook/?path=/story/components-nci-autocomplete--default)

File: yarn.lock [hunk 1]
```diff
@@ -8574,11 +8574,6 @@ array-flatten@^2.1.2:
   resolved "https://registry.yarnpkg.com/array-flatten/-/array-flatten-2.1.2.tgz#24ef80a28c1a893617e2149b0c6d0d788293b099"
   integrity sha512-hNfzcOV8W4NdualtqBFPyVO+54DSJuZGY9qT4pRroB6S9e3iiido2ISIC5h9R2sPJ8H3FHCIiEnsv1lPXO3KtQ==
 
-array-flatten@^3.0.0:
```
Comment (olitharp-nci) [inline yarn.lock:8577]: did you `yarn install` or `lerna bootstrap` without the frozen lockfile flag? Shouldn't need any changes to yarn.lock

File: testing/ncids-css-testing/stories/uswds-native/usa-table/src/index.js [hunk 2]
```diff
@@ -0,0 +1,258 @@
+const select = require("../../uswds-core/src/js/utils/select");
```
Comment (olitharp-nci) [inline testing/ncids-css-testing/stories/uswds-native/usa-table/src/index.js:1]: You shouldn't need this file

File: testing/ncids-css-testing/stories/uswds-native/usa-table/src/usa-table.scenarios.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+		viewports: [
+			{
+				label: 'mobile',
+				width: 479,
+				height: 360,
+			},
+		],
+	},
+	{
+		label: 'usa-table sortable',
```
Comment (olitharp-nci) [inline testing/ncids-css-testing/stories/uswds-native/usa-table/src/usa-table.scenarios.js:36]: need focus/hover/active checks for `.usa-table__header__button` for all three variants

File: testing/ncids-css-testing/stories/uswds-native/usa-table/src/usa-table.stories.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+		behavior: table,
+	},
+};
+
+const Template = (args) => Component(args);
+const BorderlessTemplate = (args) => Component(args);
+const StripedTemplate = (args) => Component(args);
+const SortableTemplate = (args) => SortableComponent(args);
+
+export const Default = () => <TestCase css={css} html={Template.bind({})(DefaultContent)} />;
```
Comment (olitharp-nci) [inline testing/ncids-css-testing/stories/uswds-native/usa-table/src/usa-table.stories.js:21]: Match USWDS

```text
- Table
  - Default
  - Borderless
  - Striped
  - Sortable
  - Scrollable
    - Default
    - Striped
    - Compact
    - Compact Striped
  - Stacked
    - Default
    - Borderless
    - With Header
    - With Header Borderless
```

Comment (olitharp-nci) [reply testing/ncids-css-testing/stories/uswds-native/usa-table/src/usa-table.stories.js:21]: Scrollable missing

File: packages/ncids-css/package.json [hunk 5]
```diff
@@ -119,7 +118,6 @@
 		]
 	},
 	"uswdsExcludeComments": {
-		"//": "usa-content, usa-dark-background, usa-display, usa-intro, usa-link, usa-paragraph, usa-prose all required for uswds-typography.",
-		"usa-add-aspect": "helper mixin, not a component"
```
Comment (olitharp-nci) [inline packages/ncids-css/package.json:123]: don't delete `usa-add-aspect` comment

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Review by arcepaul — APPROVED** [review]

**Review by olitharp-nci — APPROVED** [review]
Approved - decided we will use the chop off due to the `overflow-y:hidden` as a visual representation that scrollable is working for now. We will add a backlog item to add scrollbars per story later

**Pull request merged**